src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -354,18 +355,43 @@ private bool UpdateRunSettingsIfRequired(string runsettingsXml,List<string> sour
                     bool updateFramework = IsAutoFrameworkDetectRequired(navigator);
                     bool updatePlatform = IsAutoPlatformDetectRequired(navigator);
 
-                    if (updateFramework)
+                    var inferedFramework = inferHelper.AutoDetectFramework(sources, sourceFrameworks);
+                    var chosenFramework = inferedFramework;
+                    if(updateFramework)
                     {
-                        InferRunSettingsHelper.UpdateTargetFramework(navigator,
-                            inferHelper.AutoDetectFramework(sources)?.ToString(), overwrite: true);
+                        InferRunSettingsHelper.UpdateTargetFramework(navigator, inferedFramework?.ToString(), overwrite: true);
                         settingsUpdated = true;
                     }
+                    else
+                    {
+                        chosenFramework = commandLineOptions.TargetFrameworkVersion;
+                    }
 
+                    var inferedPlatform = inferHelper.AutoDetectArchitecture(sources, sourcePlatforms);
+                    var chosenPlatform = inferedPlatform;
                     if (updatePlatform)
                     {
-                        InferRunSettingsHelper.UpdateTargetPlatform(navigator,
-                            inferHelper.AutoDetectArchitecture(sources).ToString(), overwrite: true);
-                        settingsUpdated = true;
+                        InferRunSettingsHelper.UpdateTargetPlatform(navigator, inferedPlatform.ToString(), overwrite: true);
+                    }
+                    else
+                    {
+                        chosenPlatform = commandLineOptions.TargetArchitecture;
+                    }
+
+                    string incompatiableSettingWarning = string.Empty;
+
+                    var compatibleSources = InferRunSettingsHelper.FilterCompatibleSources(chosenPlatform, chosenFramework, sourcePlatforms, sourceFrameworks, out incompatiableSettingWarning);
+
+                    if(!string.IsNullOrEmpty(incompatiableSettingWarning))
+                    {
+                        EqtTrace.Info(incompatiableSettingWarning);
+                        LoggerUtilities.RaiseTestRunWarning(this.testLoggerManager, this.testRunResultAggregator, incompatiableSettingWarning);
```

Comment:
Will this print message in designmode too?

Reply:
Yes.